app/controllers/messages_controller.rb

```diff
@@ -0,0 +1,8 @@
+class MessagesController < ApplicationController
+  def index
+    @messages = current_user.messages
+                            .where(job_id: params[:job])
+                            .where('sender_id = ? OR recipient_id = ?', params[:with], params[:with])
+                            .order(created_at: :desc)
```

Comment:
I feel like we might want to refactor this. maybe something like 
```
current_user.messages.job_conversation(job_conversation_params)
```
where the job conversation params are the job_id and with_id